1-js/05-data-types/01-primitives-methods/article.md

```diff
@@ -61,15 +61,15 @@ let str = "Hello";
 alert( str.toUpperCase() ); // HELLO
 ```
 
-Simple, right? And here's what actually happens in `str.toUpperCase()`:
+Simple, right? Here's what actually happens in `str.toUpperCase()`:
 
-1. The string `str` is a primitive. So in the moment of accessing its property a special object is created that both knows the value of the string and has useful methods, like `toUpperCase()`.
+1. The string `str` is a primitive. So in the moment of accessing its property, a special object is created that knows the value of the string, and has useful methods, like `toUpperCase()`.
 2. That method runs and returns a new string (shown by `alert`).
 3. The special object is destroyed, leaving the primitive `str` alone.
 
-So, primitives can provide methods, but they still remain lightweight.
+So primitives can provide methods, but they still remain lightweight.
 
-Of course, a JavaScript engine highly optimizes that process. Internally it may skip the creation of the extra object at all. But it must adhere to the specification and behave as if it creates one.
+The JavaScript engine highly optimizes this process. It may even skip the creation of the extra object at all. But it must still adhere to the specification and behave as if it creates one.
```

Comment:
I put the word "internally" to emphasize that the change is purely internal, not visible from the outside. Is it bad here?

Reply:
These seem equivalent to me. "Internally" is more descriptive, but the following sentence implies that something is happening "behind the scenes".

Reply:
ok